### PR TITLE
Update links v3 -> v4

### LINKED ISSUE #663: Broken link in documentation
The documentation page: https://ignitionrobotics.org/api/gazebo/4.0/resources.html has a broken link in the section *Ignition GUI plugins*. The link ` GUI config files` points to the 3.0 version, which returns a 404 error. Changing the link to
https://ignitionrobotics.org/api/gui/4.0/config.html
resolves this problem.

I don't know where the documentation lives, but if somebody provides me with a pointer, I will submit a small PR to fix it.

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
# 🦟 Bug fix

Fixes #663

## Summary
I've also checked the other links. Some GH ones were pointing to `master`; the branch doesn't exist and I assume it should be `gazebo4`. Also, I've updated all the other links to point to `4.0` instead of `3.0`.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by ahcorde** [comment]
do you mind to sign the commit ? [DCO is failing](https://github.com/ignitionrobotics/ign-gazebo/pull/664/checks?check_run_id=2020504424). There are instructions [here](https://ignitionrobotics.org/docs/all/contributing#process).

File: tutorials/resources.md [hunk 1]
```diff
@@ -31,16 +31,16 @@ System plugins may be loaded through:
     * Attached to a **model**: `<model><plugin>`
     * Attached to a **sensor**: `<sensor><plugin>`
 * Passing the shared library and class to be loaded through
-  [PluginInfo](https://ignitionrobotics.org/api/gazebo/3.0/classignition_1_1gazebo_1_1ServerConfig_1_1PluginInfo.html)
-  (within [ServerConfig](https://ignitionrobotics.org/api/gazebo/3.0/classignition_1_1gazebo_1_1ServerConfig.html))
+  [PluginInfo](https://ignitionrobotics.org/api/gazebo/4.0/classignition_1_1gazebo_1_1ServerConfig_1_1PluginInfo.html)
```
Comment (ahcorde) [inline tutorials/resources.md:34]:
```suggestion
  [PluginInfo](https://ignitionrobotics.org/api/gazebo/4.6/classignition_1_1gazebo_1_1ServerConfig_1_1PluginInfo.html)
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
The current gazebo version is 4.6. Please update the links

File: tutorials/resources.md (hunk 1 above)
Comment (FirefoxMetzger) [reply tutorials/resources.md]: @ahcorde Good catch! I updated the links that point to gazebo to 4.6 and the ones pointing to the gui to 4.2

**Comment by FirefoxMetzger** [comment]
@ahcorde There seem to be build errors (missing dependencies), but I am 99.8% certain that my changes to the documentation didn't mess with any dependencies 😄 

The contribution guidelines ask for passing tests, but I'm not sure how applicable this is here. Please advise what additions are needed for this PR to get merged.

**Review by ahcorde — APPROVED** [review]
> @ahcorde There seem to be build errors (missing dependencies), but I am 99.8% certain that my changes to the documentation didn't mess with any dependencies
> The contribution guidelines ask for passing tests, but I'm not sure how applicable this is here. Please advise what additions are needed for this PR to get merged.

This is not applicable here. Thank you for the contribution :wink:

**Review by chapulina — APPROVED** [review]

**Pull request merged**